addons/mail/models/discuss/res_partner.py

```diff
@@ -77,11 +77,29 @@ def get_mention_suggestions_from_channel(self, channel_id, search, limit=8):
                 [("channel_ids", "in", channel.id)],
             ]
         )
-        partners = self._search_mention_suggestions(domain, limit)
+        extra_domain = expression.AND([
+            [('user_ids', '!=', False)],
+            [('user_ids.active', '=', True)],
+            [('partner_share', '=', False)]
+        ])
+        if channel.group_public_id.id:
+            extra_domain = expression.AND(
+                [
+                    extra_domain,
+                    [("user_ids.groups_id", "in", channel.group_public_id.id)],
+                ]
+            )
+        partners = self._search_mention_suggestions(domain, limit, extra_domain)
         members = self.env["discuss.channel.member"].search(
             [
                 ("channel_id", "=", channel.id),
                 ("partner_id", "in", partners.ids),
             ]
         )
-        return Store(members, fields={"channel": [], "persona": []}).add(partners).get_result()
+        store = Store(members, fields={"channel": [], "persona": []})
+        store.add(channel, {"authorizedGroupId": channel.group_public_id.id or None})
```

Comment:
We are trying our best to keep same field names between python and JS, so here you can use `group_public_id`.

Also this should be in `_to_store` so this simply available when adding the channel to the store, as this uses the `_to_store` formatted data.